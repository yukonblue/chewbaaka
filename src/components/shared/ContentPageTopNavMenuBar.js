/**
 * ContentPageTopNavMenuBar.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Aug 22, 2020
 */

import React from 'react'

import { Link } from 'react-router-dom'

import logo from './assets/cheetah-conservation-fund-30-logo.svg'

import './ContentPageSharedStyles.css'

import './ContentPageTopNavMenuBar.css'

export default class ContentPageTopNavMenuBar extends React.Component {

  render() {
    return (
      <div className="ContentPageTopNavMenuBarContainer">
        {this.renderNavBarLhs()}
        {this.renderNavBarRhs()}
      </div>
    );
  }

  renderNavBarLhs() {
    return (
      <div className="ContentPageTopNavMenuBarLogoPseudoContainer">
        <a href="https://cheetah.org/">
          <img
            src={logo}
            alt="CCF logo"
            width="282"
            height="60"
          />
        </a>
      </div>
    );
  }

  renderNavBarRhs() {
    return (
      <div className="ContentPageTopNavMenuBarMenuListSidePseudoContainer">
        {this.renderNavBarMenuList()}
      </div>
    );
  }

  renderNavBarMenuList() {
    const menuItems = [];
    const menuItemNames = ["Home", "History", "Biology", "Ecology", "Future"]

    for (const [idx, val] of menuItemNames.entries()) {
      menuItems.push(
        <li
          key={idx}
          className={this.props.pageTitle === val ? "ActiveItem" : ""}
        >
          <Link to={"/" + (val.toLowerCase() === "home" ? "" : val.toLowerCase())} key={idx}>
            {val}
          </Link>
        </li>
      );
    }

    return (
      <nav>
        <ul className="ContentPageTopNavMenuBarMenuList">
          {menuItems}
        </ul>
      </nav>
    );
  }
}
