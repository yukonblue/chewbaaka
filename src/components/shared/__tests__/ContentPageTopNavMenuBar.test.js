/**
 * ContentPageTopNavMenuBar.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Aug 25, 2021
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageTopNavMenuBar from '../ContentPageTopNavMenuBar'

import RouterWrapped from '../../../testing/RouterWrapped'

test('renders ContentPageTopNavMenuBar component', () => {
  render(
    RouterWrapped(
      <ContentPageTopNavMenuBar
        pageTitle="Ecology"
      />
    )
  );

  const expectedMenuItemNames = ["Home", "History", "Biology", "Ecology", "Future"];

  const getExpectedMenuItemHref = (menuItemName) => ("/" + (menuItemName.toLowerCase() === "home" ? "" : menuItemName.toLowerCase() + "/"));

  for (const [_, menuItemName] of expectedMenuItemNames.entries()) {
    const menuItemElement = screen.getByText(menuItemName);
    expect(menuItemElement).toBeInTheDocument();
    expect(menuItemElement).toHaveAttribute('href', getExpectedMenuItemHref(menuItemName));
  }
});

test('ContentPageTopNavMenuBar component snapshot', () => {
  const tree = renderer
    .create(
      RouterWrapped(
        <ContentPageTopNavMenuBar
          pageTitle="Future"
        />
      )
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
