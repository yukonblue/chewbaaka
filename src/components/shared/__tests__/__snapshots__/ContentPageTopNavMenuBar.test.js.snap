// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageTopNavMenuBar component snapshot 1`] = `
<div
  className="ContentPageTopNavMenuBarContainer"
>
  <div
    className="ContentPageTopNavMenuBarLogoPseudoContainer"
  >
    <a
      href="https://cheetah.org/"
    >
      <img
        alt="CCF logo"
        height="60"
        src="cheetah-conservation-fund-30-logo.svg"
        width="282"
      />
    </a>
  </div>
  <div
    className="ContentPageTopNavMenuBarMenuListSidePseudoContainer"
  >
    <nav>
      <ul
        className="ContentPageTopNavMenuBarMenuList"
      >
        <li
          className=""
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className=""
        >
          <a
            href="/history/"
            onClick={[Function]}
          >
            History
          </a>
        </li>
        <li
          className=""
        >
          <a
            href="/biology/"
            onClick={[Function]}
          >
            Biology
          </a>
        </li>
        <li
          className=""
        >
          <a
            href="/ecology/"
            onClick={[Function]}
          >
            Ecology
          </a>
        </li>
        <li
          className="ActiveItem"
        >
          <a
            href="/future/"
            onClick={[Function]}
          >
            Future
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
